source/agora/network/DNSResolver.d

```diff
@@ -101,6 +101,11 @@ public final class VibeDNSResolver : DNSResolver
                 auto answer = response.deserializeFull!Message();
                 log.trace("Got response from '{}' for '{}' : {}",
                           peer.address, msg, answer);
+
+                // Message is truncated, retry over TCP
+                if (answer.header.TC == true)
+                    return this.queryTCP(msg);
+
```

Comment:
I think that should be handled at the higher level. Having something called `queryUDP` perform a TCP query is surprising. And since `answer.header.RCODE` will be `NoError` (correct me if I'm mistaken), the higher level `query` can handle the retry.

Reply:
I need to update return type for `queryUDP` and `queryTCP` then; it would be good to have another look after that.

Reply:
I've updated, last 2 commits can be reviewed